rskj-core/src/integrationTest/java/co/rsk/EtherSwapTest.java

```diff
@@ -0,0 +1,318 @@
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Stream;
+
+public class EtherSwapTest {
+    private static final int LOCAL_PORT = 4444;
+
+    private static final CallTransaction.Function CALL_LOCK_FUNCTION = CallTransaction.Function.fromSignature(
+            "lock",
+            new String[]{"bytes32", "address", "uint256"},
+            new String[]{}
+    );
+    private static final CallTransaction.Function CALL_CLAIM_FUNCTION = CallTransaction.Function.fromSignature(
+            "claim",
+            new String[]{"bytes32", "uint256", "address", "uint256"},
+            new String[]{}
+    );
+    private final ObjectMapper objectMapper = new ObjectMapper();
+    private String buildLibsPath;
+    private String jarName;
+    private String databaseDir;
+    @TempDir
+    private Path tempDir;
+
+    private String[] baseArgs;
+    private String strBaseArgs;
+    private String baseJavaCmd;
+
+    @BeforeEach
+    public void setup() throws IOException {
+        String projectPath = System.getProperty("user.dir");
+        buildLibsPath = String.format("%s/build/libs", projectPath);
+        String integrationTestResourcesPath = String.format("%s/src/integrationTest/resources", projectPath);
+        String rskConfFile = String.format("%s/integration-test-rskj.conf", integrationTestResourcesPath);
+        Stream<Path> pathsStream = Files.list(Paths.get(buildLibsPath));
+        jarName = pathsStream.filter(p -> !p.toFile().isDirectory())
+                .map(p -> p.getFileName().toString())
+                .filter(fn -> fn.endsWith("-all.jar"))
+                .findFirst()
+                .get();
+        Path databaseDirPath = tempDir.resolve("database");
+        databaseDir = databaseDirPath.toString();
+        baseArgs = new String[]{"--regtest"};
+        strBaseArgs = String.join(" ", baseArgs);
+        baseJavaCmd = String.format("java %s", String.format("-Drsk.conf.file=%s", rskConfFile));
+    }
+
+    private Response lockTxRequest(String refundAddress, byte[] lockData, BigInteger amount) throws IOException {
+        String lockTxRequestContent = "[{\n" +
+                "    \"method\": \"eth_sendTransaction\",\n" +
+                "    \"params\": [{\n" +
+                "        \"from\": \"" + refundAddress + "\",\n" +
+                "        \"to\": \"0x" + Constants.regtest().getEtherSwapContractAddress() + "\",\n" +
+                "        \"data\": \"0x" + ByteUtil.toHexString(lockData) + "\",\n" +
+                "        \"value\": \"" + HexUtils.toQuantityJsonHex(amount.longValue()) + "\",\n" +
+                "        \"gas\": \"0xc350\",\n" +
+                "        \"gasPrice\": \"0x1\"\n" +
+                "    }],\n" +
+                "    \"id\": 1,\n" +
+                "    \"jsonrpc\": \"2.0\"\n" +
+                "}]";
```

Comment:
Could we use some cleaner pattern for JSON construction, something like a builder? This is quite diff to read.
If none is available at hand, it could be as simple as:
```
public class JsonBuilder {
        private final ObjectMapper mapper = new ObjectMapper();
        private final ObjectNode rootNode = mapper.createObjectNode();

        public JsonBuilder set(String key, String value) {
            rootNode.put(key, value);
            return this;
        }

        public JsonBuilder set(String key, int value) {
            rootNode.put(key, value);
            return this;
        }

        public JsonBuilder set(String key, ObjectNode value) {
            rootNode.set(key, value);
            return this;
        }

        public JsonBuilder set(String key, ArrayNode value) {
            rootNode.set(key, value);
            return this;
        }

        public String build() {
            return rootNode.toString();
        }

        public String buildArray() {
            return "[" + rootNode.toString() + "]";
        }
    }
```

so then we can use it as so:

```suggestion
String lockTxRequestContent = new JsonBuilder().set("method", "eth_sendTransaction")
                .set("params", new JsonBuilder()
                        .set("from", refundAddress)
                        .set("to", "0x" + Constants.regtest().getEtherSwapContractAddress())
                        .set("data", "0x" + ByteUtil.toHexString(lockData))
                        .set("value", HexUtils.toQuantityJsonHex(amount.longValue()))
                        .set("gas", "0xc350")
                        .set("gasPrice", "0x1")
                        .buildArray())
                .set("id", 1)
                .set("jsonrpc", "2.0").buildArray();
```

Reply:
I think we can do this in another ticket since the change will just affect the readability of the code and we could also refactor other instances were there are other Json variables with the same format.